<details><summary>So far so good.</summary>

```
export PATH_TMP_SUCO=~/.local/share/tmp_suco && \
  export PATH_VENV=~/.local/share/tmp_suco/venv && \
  sudo apt update && sudo apt install python3-venv && \
  mkdir -p $PATH_TMP_SUCO && cd $PATH_TMP_SUCO && python3 -m venv $PATH_VENV && source $PATH_VENV/bin/activate
$ export SUCO_INSTALLER=/tmp/suco.py &&  \
  export VERSION=develop &&  \
  sudo apt update && sudo apt install -y curl git python3 python3-pip &&  \
  curl --output $SUCO_INSTALLER https://raw.githubusercontent.com/kinu-garage/hut_10sqft/$VERSION/hut_10sqft/hut_10sqft/suco_installer.py && chmod 755 $SUCO_INSTALLER && \
  export GIT_CONFIG_GLOBAL=''; export GIT_CONFIG_SYSTEM=''; $SUCO_INSTALLER --git_branch $VERSION
$ export PATH_TMP_WS=~/.local/share/tmp_suco/colconws &&  \
  export HOST_NAME=130s-c14-morph &&  \
  export OS_DISTRO=ChromeOS &&  \
  source $PATH_TMP_WS/install/setup.bash && \
  cd $PATH_TMP_WS/src/hut_10sqft/hut_10sqft && python -m pip install setuptools && python setup.py install && cd $PATH_TMP_WS
```
</details>

Except I found `Ubuntu` notation. 
```
$ cat /etc/apt/sources.list.d/docker.list 
deb https://apt.dockerproject.org/repo ubuntu-bookworm main
```
I don't even know why we've set up Docker on ChromeOS, where Debian runs in a container. So removed the apt source:
```
sudo rm /etc/apt/sources.list.d/docker.list
```

Then got an error w/the next and final cmd:
```
$ suco --hostname $HOST_NAME --os_distro $OS_DISTRO --conf_repo_version $VERSION && \
  echo "Removing a temp colcon ws dir".; rm -fr $PATH_TMP_WS

:

Latest ROS apt source version: '1.1.0",'
Traceback (most recent call last):
  File "/home/gm130s/.local/share/tmp_suco/colconws/install/hut_10sqft/bin/suco", line 33, in <module>
    sys.exit(load_entry_point('hut-10sqft', 'console_scripts', 'suco')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gm130s/.local/share/tmp_suco/colconws/build/hut_10sqft/hut_10sqft/suco_main.py", line 153, in main
    comp_setup.run()
  File "/home/gm130s/.local/share/tmp_suco/colconws/build/hut_10sqft/hut_10sqft/suco_main.py", line 135, in run
    _os_builder.run(_host_cfg,
  File "/home/gm130s/.local/share/tmp_suco/colconws/build/hut_10sqft/hut_10sqft/comp_debian.py", line 194, in run
    super().run(host_config, conf_repo_remote, conf_base_path)
  File "/home/gm130s/.local/share/tmp_suco/colconws/build/hut_10sqft/hut_10sqft/abst_comp_setup.py", line 417, in run
    self.setup_rosdep_and_run(self._args_in.path_temp_colconws, init_rosdep=True)
  File "/home/gm130s/.local/share/tmp_suco/colconws/build/hut_10sqft/hut_10sqft/comp_ubuntu.py", line 304, in setup_rosdep_and_run
    self.exec_rosdep_update(path_ws, pkg_rosdep, init_rosdep)
  File "/home/gm130s/.local/share/tmp_suco/colconws/build/hut_10sqft/hut_10sqft/comp_debian.py", line 103, in exec_rosdep_update
    self.setup_ros_installer_src()
  File "/home/gm130s/.local/share/tmp_suco/colconws/build/hut_10sqft/hut_10sqft/comp_ubuntu.py", line 293, in setup_ros_installer_src
    self.set_ros_apt_source()
  File "/home/gm130s/.local/share/tmp_suco/colconws/build/hut_10sqft/hut_10sqft/comp_ubuntu.py", line 240, in set_ros_apt_source
    _ubu_codename = _env_vars_os_release["UBUNTU_CODENAME"]
                    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
KeyError: 'UBUNTU_CODENAME'
```

Turned out this was just an operational glitch (setting Ubuntu as `OSTYPE`), fixed it.

<details><summary>The run passed, although many deb pkgs didn't seem to get installed.</summary>

```
$ export PATH_TMP_WS=~/.local/share/tmp_suco/colconws &&  \
  export HOST_NAME=130s-c14-morph &&  \
  export OS_DISTRO=ChromeOS &&  \
  source $PATH_TMP_WS/install/setup.bash && \
  cd $PATH_TMP_WS/src/hut_10sqft/hut_10sqft && python -m pip install setuptools && python setup.py install && cd $PATH_TMP_WS

:

Installing rst2xetex script to /home/gm130s/.local/share/tmp_suco/venv/bin
Installing rst2xml script to /home/gm130s/.local/share/tmp_suco/venv/bin

Using /home/gm130s/.local/share/tmp_suco/venv/lib/python3.11/site-packages
Searching for six==1.17.0
Best match: six 1.17.0
Adding six 1.17.0 to easy-install.pth file

Using /home/gm130s/.local/share/tmp_suco/venv/lib/python3.11/site-packages
Finished processing dependencies for hut-10sqft==0.4.0
(venv) gm130s@penguin:~/.local/share/tmp_suco/colconws$ ll
total 0
drwxr-xr-x 1 gm130s gm130s 404 Jan 22 10:53 install/
drwxr-xr-x 1 gm130s gm130s  92 Jan 22 10:53 build/
drwxr-xr-x 1 gm130s gm130s  36 Jan 22 10:53 ./
drwxr-xr-x 1 gm130s gm130s 112 Jan 22 10:53 log/
drwxr-xr-x 1 gm130s gm130s  20 Jan 22 10:53 src/
drwxr-xr-x 1 gm130s gm130s  24 Jan 22 10:53 ../
```
```
$ suco --hostname $HOST_NAME --os_distro $OS_DISTRO --conf_repo_version $VERSION && \
  echo "Removing a temp colcon ws dir".; rm -fr $PATH_TMP_WS

:

loading cached data source:
        https://raw.githubusercontent.com/ros/rosdistro/master/rolling/distribution.yaml
        /home/gm130s/.ros/rosdep/sources.cache/10be71ecf140657927f11aa700ec034e8f4759d7.pickle
loaded 11 sources
5 sources match current tags
resolve_all: resource [hut_10sqft_lib] requires rosdep keys [ament_copyright, ament_flake8, ament_pep257, python3-pytest]
resolve_all: resource [hut_10sqft] requires rosdep keys [ack, apt-transport-https, ca-certificates, emacs, emacs-mozc, freecad, gimp, git, gnuplot, gnuplot-x11, hut_10sqft_lib, iftop, imagemagick, ipython3, libwebp-tools, meld, mesa-utils, meshlab, mozc-server, ntp, openjdk-7-jre, python3-bloom, python3-empy, python3-git, python3-pip, python3-rocker, python3-rosdep, python3-vcstool, rpl, ack, debtree, nmap, openssh-client, openssh-server, texlive-fonts-recommended, texlive-latex-base, tmux, tree, xdotool, ament_copyright, ament_flake8, ament_pep257, python3-pytest]
rosdep key 'hut_10sqft_lib' is in the catkin workspace, skipping.
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies (ROS distro is not set. Make sure `ROS_DISTRO` environment variable is set, or use `--rosdistro` option to specify the distro, e.g. `--rosdistro indigo`):
hut_10sqft: Cannot locate rosdep definition for [rpl]
Continuing to install resolvable dependencies...
        Issue #4: Report: Package installation status:

- Packages found installed: []
- The following pkgs didn't get installed:
        - Failed to get apt-cache policy for 'gitk'. Error: The install candidate of the pkg 'gitk' found on this OS, but 'apt-cache policy' didn't find it installed.
        - Failed to get apt-cache policy for 'kazam'. Error: The install candidate of the pkg 'kazam' found on this OS, but 'apt-cache policy' didn't find it installed.
        - Failed to get apt-cache policy for 'libreoffice-drawlocate'. Error: 'libreoffice-drawlocate' is not found on this OS, even as an install candidate. Check if it is really available.
        - Failed to get apt-cache policy for 'lvm2'. Error: The install candidate of the pkg 'lvm2' found on this OS, but 'apt-cache policy' didn't find it installed.
        - Failed to get apt-cache policy for 'mpv'. Error: The install candidate of the pkg 'mpv' found on this OS, but 'apt-cache policy' didn't find it installed.
        - Failed to get apt-cache policy for 'smartmontools'. Error: The install candidate of the pkg 'smartmontools' found on this OS, but 'apt-cache policy' didn't find it installed.
        - Failed to get apt-cache policy for 'vulkan-tools'. Error: The install candidate of the pkg 'vulkan-tools' found on this OS, but 'apt-cache policy' didn't find it installed.
        - Failed to get apt-cache policy for 'virtualbox-guest-additions-iso'. Error: 'virtualbox-guest-additions-iso' is not found on this OS, even as an install candidate. Check if it is really available.
        - Failed to get apt-cache policy for 'virtualbox-qt'. Error: The install candidate of the pkg 'virtualbox-qt' found on this OS, but 'apt-cache policy' didn't find it installed.

        Issue #5: Configs of the visual of the Terminal needs to be done manually  either using (recommended) Terminal's GUI on 'Preference' or using `dconf` and the premade config files, which you can find in '/home/gm130s/.config/hut_10sqft/hut_10sqft/config/dconf'. See https://github.com/kinu-garage/hut_10sqft/issues/174#issuecomment-3003542573
        Issue #6: An error regarding socket occurred while verifying ssh server operation: [Errno 111] Connection refused
        Issue #7: [Errno 17] File exists: '/home/gm130s/data'
        Issue #8: [Errno 17] File exists: '/home/gm130s/link'
```
</details>

--- a/.github/workflows/python-app.yml
+++ b/.github/workflows/python-app.yml
@@ -18,7 +18,7 @@ jobs:
     strategy:
       fail-fast: false  # Allows other jobs in the matrix to continue if one fails
       matrix:
-        os: [ubuntu-24.04, macos-13]
+        os: [ubuntu-24.04, macos-14]
 #        os: [debian-11, ubuntu-24.04, macos-11]
     runs-on: ${{ matrix.os }}
 

--- a/hut_10sqft/doc/index.md
+++ b/hut_10sqft/doc/index.md
@@ -23,8 +23,8 @@ Prerequisite:
      export OS_DiSTRO=Ubuntu &&  \
      source $PATH_TMP_WS/install/setup.bash && \
      cd $PATH_TMP_WS/src/hut_10sqft/hut_10sqft && python -m pip install setuptools && python setup.py install && cd $PATH_TMP_WS
-   $ suco --hostname $HOST_NAME --os_distro $OS_DISTRO --conf_repo_version $VERSION && \
-     echo "Removing a temp colcon ws dir".; rm -fr $PATH_TMP_WS
+   $ suco --hostname $HOST_NAME --os_distro $OS_DISTRO --conf_repo_version $VERSION
+   $ echo "Removing a temp colcon ws dir".; rm -fr $PATH_TMP_WS
    ```
    Note as of v0.3.1, when re-running SUCO, running the whole set of lines, instead of resuming at certain line in the middle, is recommended.
 

